rocksdbserver/rocksdbserver.py

```diff
@@ -275,42 +189,97 @@ def list_values(self):
         _iter.seek_to_first()
         return list(self.unpackfn(x) for x in _iter)
 
-    @ensurenewiter
-    def iter_keys(self, name=None, reverse=False):
-        self.iters[name] = self.iter_klass(self, type='keys', reverse=reverse)
-        return name
-
-    @ensurenewiter
-    def iter_values(self, name=None, reverse=False):
-        self.iters[name] = self.iter_klass(self, type='values', reverse=reverse)
-        return name
 
-    @ensurenewiter
-    def iter_items(self, name=None, reverse=False):
-        self.iters[name] = self.iter_klass(self, type='items', reverse=reverse)
-        return name
+    def _configure_iterator(self, iterator, seek_to=None, reverse=False):
+        """
+        configures the iterator by seeking to the right position and
+        reversing if required.
+        """
+        if seek_to is not None:
+            iterator.seek(seek_to)
+        elif not reverse:
+            iterator.seek_to_first()
+        else:
+            iterator.seek_to_last()
 
-    def list_iters(self):
-        return self.iters.keys()
+        if reverse:
+            iterator = reversed(iterator)
 
-    def close_iter(self, _iter):
-        del self.iters[_iter]
+        return iterator
 
-    @ensureiter
-    def iter_get(self, _iter, num=Iterator.NUM_RECORDS):
-        return _iter.get(num)
+    def iter_keys(self, seek_to=None, reverse=False, regex=None):
+        """
+        iterates through the keys in the database.
+        `seek_to` - seeks to the given position if specified.
+            defaults to the first record if reverse is False.
+            defaults to the last record if reverse is True.
 
-    @ensureiter
-    def iter_seek(self, _iter, key):
-        return _iter.seek(key)
+        `regex` - returns only keys that match the regex.
+        """
 
-    @ensureiter
-    def iter_seek_to_first(self, _iter):
-        return _iter.seek_to_first()
+        if regex is not None:
+            regex = re.compile(regex)
 
-    @ensureiter
-    def iter_seek_to_last(self, _iter):
-        return _iter.seek_to_last()
+        _iter = self.rdb.iterkeys()
+        _iter = self._configure_iterator(_iter, seek_to=seek_to, reverse=reverse)
+        for key in _iter:
+            if regex is not None and not regex.match(key):
+                continue
+
+            yield key
+
+    def iter_items(self, seek_to=None, reverse=False, regex=None, jq_filter=None):
```

Comment:
@algrebe Instead of `jq_filter`, can you please use `value_filter` or something similar?